### ISSUE TITLE
In GPUI, when using WindowKind::PopUp with the background set to WindowBackgroundAppearance::Transparent, a border is still displayed.

### ISSUE THREAD

**Issue by mllw** [zed-industries/zed#61508]
### Reproduction steps

<img width="258" height="238" alt="Image" src="https://github.com/user-attachments/assets/dace3a85-614d-449e-99d8-3304e4a98e2e" />
<img width="307" height="273" alt="Image" src="https://github.com/user-attachments/assets/3d717ecb-ca3d-4633-8ea4-c92520edc56b" />

`
use gpui::{WindowOptions, prelude::*, *};

pub struct MainView;
impl Render for MainView {
    fn render(&mut self, window: &mut gpui::Window, cx: &mut Context<Self>) -> impl IntoElement {
        div()
            .bg(gpui::blue())
            .rounded(px(100.))
            .size_full()
    }
}

fn main() {
    gpui_platform::application().run(|app|{
        let window_bounds = WindowBounds::centered(size(px(200.), px(200.)), app);
        app.spawn(async move |app|{
            app.open_window(WindowOptions {
                        titlebar: None,
                        window_min_size: Some(gpui::size(px(200.), px(200.))),
                        window_bounds: Some(window_bounds),
                        window_background: WindowBackgroundAppearance::Transparent,
                        kind: WindowKind::PopUp,
                        window_decorations: None,
                        is_resizable: false,
                        is_minimizable: false,
                        is_movable: true,
                        ..Default::default()
                    }, |window,app|{                        
                        app.new(|app|{MainView})
                    }).expect("");
        })
        .detach();
    });
}

`

### Current vs. Expected behavior

dont show window border or how to hiden window border

### Zed version and system specs

git log
commit f9c994796ad4341649d7b8664edbdfaae8bebd5d (HEAD -> master)

### If you are using WSL on Windows, what flavor of Linux are you using?

None

**Comment by mllw** [comment]
OS: Windows 11 家庭版 中文版 25H2

**Comment by lamersc** [comment]
[61508_repro.zip](https://github.com/user-attachments/files/30509311/61508_repro.zip)

Can confirm this is **reproducible**, though I am not certain if this is a bug or a missing feature, as having the option to keep the border around a transparent window still feels like it could be useful; nevertheless, I have attached a reproduction video and project.

### Environment
- OS: macOS 26.5.2
- Arch: arm64
- Reproduced on:
  ```
  Zed 1.12.1 
  Commit: 2a37601c02a32b22e7700835c04b89ff75ffcd5d
  Version: 1.12.1+stable.330.2a37601c02a32b22e7700835c04b89ff75ffcd5d
  ```

### Reproduction
1. Open the attached project inside Zed and make sure you have rust installed.
2. Note inside `src/main.rs` that `WindowOptions.window_background` is set to `WindowBackgroundAppearance::Transparent`
3. Run `cargo run` inside Zeds terminal at the projects root.
4. Observe that while the background for the window is transparent, the border still appears.
5. Observe after that there is no platform agnostic way to hide this border (though it can be done manually e.g. by reaching out to `objc2-app-kit` and `objc2` on macOS).

### Expectation
It is expected that there should be some way to completely hide all attributes of a window such that only the content being drawn directly to it is visible on screen.

### Video
https://github.com/user-attachments/assets/02f70749-9d08-4239-90a8-d2e3ef273502

cc: @yeskunall @AidanV

**Comment by ppmpreetham** [comment]
Can reproduce the same on Windows
